src/wp-includes/class-wp-block.php

```diff
@@ -106,18 +131,71 @@ class WP_Block {
 	 * @param array                  $block             Array of parsed block properties.
 	 * @param array                  $available_context Optional array of ancestry context values.
 	 * @param WP_Block_Type_Registry $registry          Optional block type registry.
+	 * @param array                  $options {
+	 *   Optional options object.
+	 *
+	 *   @type bool $is_eager When false, none of the derived block properties are set until `render()` is called. Defaults to true. See `set_derived_properties()` for more info.
+	 * }
+	 */
+	public function __construct( $parsed_block, $available_context = array(), $registry = null, $options = null ) {
+		$options = wp_parse_args(
+			$options,
+			array(
+				'is_eager' => true,
+			)
+		);
+
+		$this->parsed_block      = $parsed_block;
+		$this->available_context = $available_context;
+		$this->registry          = $registry ? $registry : WP_Block_Type_Registry::get_instance();
+		$this->options           = $options;
+
+		if ( $this->options['is_eager'] ) {
+			$this->set_derived_properties();
+		}
+	}
```

Comment:
Nit: `$args` instead of `$options` for consistency with the majority of the code base.

Reply:
Hm, I was trying to be consistent with `WP_Block::render()` which already has `$options`. Reckon I should change both? Or just the constructor's? Or neither?